### DISCUSSION TITLE
Does someone know if we can check variable types in eslint custom rules ?

### DISCUSSION THREAD

**Discussion by zerah54** [eslint/eslint#14938] [Q&A]
Hi there !

I'm trying to improve my project, in which I've some touchy cases I would like to highlight in my Gitlab pipelines. 

As an example, I want to check if part of "===" operator is an object or a specific type, and if it is, throw an error. I've a custom rule, which find the operator, but I'm not able to find the type of left and right parts. 

Maybe someone know if that's possible and could explain to me ?

Many thanks !

**Comment by bmish** [comment]
Have you checked out this site? https://astexplorer.net/

Enter your code sample, and then you can see corresponding AST, which will include nodes to represent the variables on each side of the operator.

**Reply by zerah54** [reply]
yes, I tried to get variables typings in ast from this site, but it doesn't seems to be part of the AST. I find a "type" which is for example "Identifier" but that's not the type in my code, only the one in tree :(

**Reply by bmish** [reply]
Oh I see. While you can get the types of any literals in the expression directly from the tree (i.e. with `x === 123`), you will want to use either of these two helper functions to get the value of a variable:
* [eslint-utils - getStaticValue](https://eslint-utils.mysticatea.dev/api/ast-utils.html#getstaticvalue) - gets the value of the variable if it can be determined statically (i.e. it would return `123` for `x` in this situation: `const x = 123; x === 123`)
* [eslint-utils - findVariable](https://eslint-utils.mysticatea.dev/api/scope-utils.html#findvariable) - gets the node where the variable is defined which you could then try to check the type of

**Reply by zerah54** [reply]
Many thanks, I finally found the type of my variable (MyType).

Now, my issue is that this class (MyType) extends another one (CustomType), and I need to get the ancestor (CustomType).

```
export class CustomType {
}

export class MyType extends CustomType {
}
```

I tried to find a reference to CustomType in the variable but didn't found.
I try to list all classes which extend CustomType so that I could compare the retreived type to the list, but they are in multiple files and I'm not able to read them :(

**Reply by aladdin-add** [reply]
if you are writing type-related rules, you need to checkout https://github.com/typescript-eslint/typescript-eslint, as eslint does not know the typings.
